starter-core/src/main/java/io/micronaut/starter/feature/opentelemetry/OpenTelemetryAnnotations.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.opentelemetry;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.build.dependencies.MicronautDependencyUtils;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class OpenTelemetryAnnotations implements OpenTelemetryFeature {
```

Comment:
According to our docs, I think the annotation processor is needed for the mappers. 

https://micronaut-projects.github.io/micronaut-tracing/latest/guide/#annotations

Reply:
Possibly, the test project I used to create the GCP guide doesn't need it to build or run. Am using Java 17 and it might be needed with older versions.